### PR TITLE
[DNM]support single delete

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Support single delete in `DeleteRequest `.
@siddontang @disksing PTAL

### CODE REVIEW

File: proto/raft_cmdpb.proto [hunk 1]
```diff
@@ -25,8 +25,9 @@ message PutResponse {
 }
 
 message DeleteRequest {
-    optional string cf  = 1;
-    optional bytes  key = 2;
+    optional string cf            = 1;
+    optional bytes  key           = 2;
+    optional bool   single_delete = 3;
```
Comment (siddontang) [inline proto/raft_cmdpb.proto]: why do we need this in protocol?

Comment (zhangjinpeng1987) [inline proto/raft_cmdpb.proto]: When apply raft log in peers, we should know the delete type.

**Comment by siddontang** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request closed without merging**